spec/features/search/tool_results_sorting_spec.rb

```diff
@@ -0,0 +1,172 @@
+        end
+
+        it 'sorts the results by Name Desc' do
+          within '#search-results tbody tr:nth-child(1)' do
+            expect(page).to have_content('ZZebra Tool Name')
+          end
+        end
+      end
+    end
+
+    context 'when sorting by Long Name' do
+      before do
+        click_on 'Sort by Long Name Asc'
+      end
+
+      it 'displays the correct search param' do
+        expect(page).to have_collection_search_query(nil, 'Sort Key: Long Name Asc')
+      end
+
+      it 'sorts the result by Long Name Asc' do
+        within '#search-results tbody tr:nth-child(1)' do
+          expect(page).to have_content('.. Agouti Tool Long Name')
+        end
+      end
+
+      context 'when sorting again' do
+        before do
+          click_on 'Sort by Long Name Desc'
+        end
+
+        it 'displays the correct search param' do
+          expect(page).to have_collection_search_query(nil, 'Sort Key: Long Name Desc')
+        end
+
+        it 'sorts the results by Long Name Desc' do
+          within '#search-results tbody tr:nth-child(1)' do
+            expect(page).to have_content('ZZebra Tool Long Name')
+          end
+        end
+      end
+    end
+
+    context 'when sorting by Provider' do
+      before do
+        click_on 'Sort by Provider Asc'
+      end
+
+      it 'displays the correct search param' do
+        expect(page).to have_collection_search_query(nil, 'Sort Key: Provider Asc')
+      end
+
+      it 'sorts the result by Provider Asc' do
+        within '#search-results tbody tr:nth-child(1) td:nth-child(3)' do
+          # we are only checking the correct provider, in case other tools
+          # have been published to the provider
+          expect(page).to have_content('LARC')
+        end
+      end
+
+      context 'when sorting again' do
+        before do
+          click_on 'Sort by Provider Desc'
+        end
+
+        it 'displays the correct search param' do
+          expect(page).to have_collection_search_query(nil, 'Sort Key: Provider Desc')
+        end
+
+        it 'sorts the results by Provider Desc' do
+          within '#search-results tbody tr:nth-child(1) td:nth-child(3)' do
+            # we are only checking the correct provider, in case other Tools
+            # have been published to the provider
+            expect(page).to have_content('SEDAC')
+          end
+        end
+      end
+    end
+
+    context 'when sorting by Last Modified' do
+      before do
+        click_on 'Sort by Last Modified Asc'
+      end
+
+      it 'displays the correct search param' do
+        expect(page).to have_collection_search_query(nil, 'Sort Key: Last Modified Asc')
+      end
+      # we cannot deterministically test sorting results by Last Modified Asc
+      # because we cannot modify the tool revision date, and there may be
+      # Tools published to providers that are not removed by reset_provider
```

Comment:
should there be a comment that we can revisit after the CMR ticket?

Reply:
This is a comment carried through from the V/S tests. I left it in because I think it will still be true when reset providers works.